domain/application/service/service.go

```diff
@@ -470,16 +464,52 @@ func (s *WatchableService) WatchApplicationExposed(ctx context.Context, name str
 		eventsource.PredicateFilter(
 			exposedToSpaces,
 			changestream.All,
-			func(s string) bool {
-				return s == uuid.String()
-			},
+			eventsource.EqualsPredicate(uuid.String()),
 		),
 		eventsource.PredicateFilter(
 			exposedToCIDRs,
 			changestream.All,
-			func(s string) bool {
-				return s == uuid.String()
-			},
+			eventsource.EqualsPredicate(uuid.String()),
+		),
+	)
+}
+
+// WatchUnitForUniterChanged watches for some specific changes to the unit with
+// the given name. The watcher will emit a notification when there is a change to
+// the unit's inherent properties, it's subordinates or it's resolved mode.
+//
+// If the unit does not exist an error satisfying [applicationerrors.UnitNotFound]
+// will be returned.
+//
+// These tables are selected since they provide coverage for the events the uniter
+// watches for using the Watch agent facade method.
+//
+// TODO(jack-w-shaw): This watcher only exists to maintain backwards compatibility
+// of the uniter agent facade. Specifically, version 20 of the facade implements
+// a Watch endpoint, which watches for. Once we no longer need to support facade 20,
```

Comment:
incomplete sentence "Specifically, version 20 of the facade implements a Watch endpoint, which watches for." For what?

Reply:
Done!